File: base/src/main/vsdk/toolkit/common/Vertex2D.java
package vsdk.toolkit.common;
import java.io.Serial;

public class Vertex2D extends FundamentalEntity
{
    @Serial private static final long serialVersionUID = 20090816L;

    public double x;
    public double y;
    public ColorRgb color;

    public Vertex2D(double x, double y)
    {
        this.x = x;
        this.y = y;
        this.color = new ColorRgb();
    }

    public Vertex2D(double x, double y, double r, double g, double b)
    {
        this.x = x;
        this.y = y;
        this.color = new ColorRgb(r, g, b);
    }

    @Override
    public String toString()
    {
        String msg = "<" + VSDK.formatDouble(x) + ", " +
            VSDK.formatDouble(y) + ">";
        return msg;
    }
}

File: cpp/base/src/main/vsdk/toolkit/environment/geometry/elements/Vertex2D.cpp
#include "vsdk/toolkit/environment/geometry/elements/Vertex2D.h"
#include "vsdk/toolkit/common/VSDK.h"

Vertex2D::Vertex2D(double inX, double inY) : x(inX), y(inY), color() {}
Vertex2D::Vertex2D(double inX, double inY, double r, double g, double b) : x(inX), y(inY), color(r, g, b) {}

std::string Vertex2D::toString() const
{
    return "<" + VSDK::formatDouble(x) + ", " + VSDK::formatDouble(y) + ">";
}
